contrib/win32/openssh/OpenSSHFixHostFilePermissions.ps1

```diff
@@ -0,0 +1,50 @@
+﻿param ([switch]$Quiet)
+Import-Module $PSScriptRoot\OpenSSHUtils.psm1 -Force -DisableNameChecking
+
+#check sshd config file
+$sshdConfigPath = join-path $PSScriptRoot "sshd_config"
+if(Test-Path $sshdConfigPath -PathType Leaf)
+{
+    Fix-HostSSHDConfigPermissions -FilePath $sshdConfigPath @psBoundParameters
+}
+else
+{
+    Write-host "----------$FilePath does not exist----------"O
+}
+ 
+#check private host keys
+$result = 'n'
+if (-not $Quiet) {
+    Do
+    {                
+        $input = Read-Host -Prompt "Did you register host private keys with ssh-agent? [Yes] Y; [No] N"    
+    } until ($input -match "^(y(es)?|N(o)?)$")
+    $result = $Matches[0]
+}
+
+if($result.ToLower().Startswith('n'))
+{
+    $warning = @"
+To keep the host private keys secure, it is recommended to register them with ssh-agent following
+steps in link https://github.com/PowerShell/Win32-OpenSSH/wiki/Install-Win32-OpenSSH.
+If you choose not to register the keys with ssh-agent, please grant sshd read access to the private host keys after run this script.
```

Comment:
Is the required?
 please grant sshd read access to the private host keys after run this script

I think the script is taking care of this? if yes then pls remove this warning message..